### PR TITLE
Refactor istio-install setup

### PR DESCRIPTION
**Pull request by howardjohn**
This brings a number of changes to simplify and make the istio install
more flexible. This removes the legacy helm setup, which is no longer
used in master (for old installs, use other branches of tools).
Additionally, it allows installing istio from any source. This is useful
for testing private builds or distributions. This also allows simple
options to configure istio with different profiles or --set flags.

Finally, the prometheus operator setup is cleaned up and merged with the
other "extras". This fixes https://github.com/istio/istio/issues/23262
in the process.

The changes here are documented in the README, please start there.

### CODE REVIEW

**Comment by howardjohn** [comment]
For what its worth, I had this setup forked for many months and it has been working very well. Trying to upstream now that i have some time so others can (hopefully) benefit from this as well

File: perf/benchmark/run_benchmark_job.sh [hunk 1]
```diff
@@ -165,23 +165,21 @@ source "${ROOT}/../bin/setup_cluster.sh"
...
 fi
 
 # different branch tag resides in dev release directory like /latest, /1.4-dev, /1.5-dev etc.
-TAG=$(curl "https://storage.googleapis.com/istio-build/dev/${BRANCH}")
-echo "Setup istio release: $TAG"
+INSTALL_VERSION=$(curl "https://storage.googleapis.com/istio-build/dev/${BRANCH}")
+echo "Setup istio release: ${INSTALL_VERSION}"
 # TAG is of the form like "1.5-alpha.sha"
 # shellcheck disable=SC2155
 export GIT_SHA=$(echo "$TAG" | cut -f3 -d.)
```
Comment (carolynhu) [inline perf/benchmark/run_benchmark_job.sh:179]: Here the `TAG` should be updated to `INSTALL_VERSION`

Comment (carolynhu) [reply perf/benchmark/run_benchmark_job.sh:179]: GIT_SHA actually is not used anywhere, we can delete it.

**Pull request merged**